Detection integration

dpthibodeau opened the pull request:
Revision of detector / embedding model pipeline. This branch has well-functioning use of retinaface/ArcFace combo.

bertiesh reviewed (commented):
Since I noticed you didn't actually change the part: if embedding is not None:
                        #if embedding_results:
                            #embedding = embedding_results[0]["embedding"]
                            face_embeddings.append({
                                "image_path": path_str,
                                "embedding": embedding,
                                "bbox": [region["x"], region["y"], region["w"], region["h"]],
                                "confidence": region["confidence"]
                            })
Is it possible to not merge all the hard-coded things and leave config, face_representation.py and interface.py the way it is and move the whole detection and getting embeddings to a new file with represent function? Plus, I don't think you have commented/deleted their original Deepface.represent. The returning value of the function still comes from deepface results.

dpthibodeau commented:
> Since I noticed you didn't actually change the part: if embedding is not None: #if embedding_results: #embedding = embedding_results[0]["embedding"] face_embeddings.append({ "image_path": path_str, "embedding": embedding, "bbox": [region["x"], region["y"], region["w"], region["h"]], "confidence": region["confidence"] }) Is it possible to not merge all the hard-coded things and leave config, face_representation.py and interface.py the way it is and move the whole detection and getting embeddings to a new file with represent function? Plus, I don't think you have commented/deleted their original Deepface.represent. The returning value of the function still comes from deepface results.

The result were and still are coming from local models and NOT from DeepFace (see the latest updates where I have completely removed the DeepFace imports)

Unless we want to entirely get rid of face_representation.py, I don't see the purpose of not changing that file. The purpose of that function was to return the embeddings from the entire pipeline. I have moved a good bit of code to other utility files, but the main results are now still being returned through that function. It is going to require more code somewhere due to the complexity of using multiple model combinations, which the original version did not have to account for locally.

I have undone the 2 modifications to interface.py, and updated the requirements to reflect no more deepface.
